python/revng/scripting/projects/project.py

```diff
@@ -0,0 +1,151 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import Any, List
+
+from revng.model import Binary, get_element_by_path, make_diff
+from revng.tupletree import TypedList
+
+
+class Project(ABC):
+    def __init__(self) -> None:
+        self.model: Binary = Binary()
+        self.last_saved_model: Binary = deepcopy(Binary)
+        self.pipeline_description: dict = {}
+        self.input_binary: str = ""
+
+    @abstractmethod
+    def import_binary(self, input_binary: str):
```

Comment:
Make sure all non-private methods are well-documented with a docstring

Reply:
Let me know if you want any of the docstrings updated, otherwise please resolve the conversation.

Reply:
There's still some places where the docstring is missing. Please double check that in a typical (e.g. tests) user interaction all the methods are properly documented